### PR TITLE
fix(suite): Add possibility of Tooltip to ButtonGroup

### PR DESCRIPTION
**Pull request by jvaclavik**
## Description

https://github.com/trezor/trezor-suite/pull/12501#discussion_r1619163965

## Related Issue

Resolve 

## Screenshots:
image

### CODE REVIEW

File: packages/components/src/components/buttons/ButtonGroup/ButtonGroup.tsx [hunk 1]
```diff
@@ -4,13 +4,15 @@ import { borders } from '@trezor/theme';
...
+import { Tooltip, TooltipProps } from '../../Tooltip/Tooltip';
 
 const Container = styled.div<{ $variant?: Exclude<ButtonVariant, 'danger'> }>`
     position: relative;
     display: flex;
     align-items: center;
 
-    > button {
+    > button,
+    div > button {
```
Comment (peter-sanderson) [inline packages/components/src/components/buttons/ButtonGroup/ButtonGroup.tsx]: :scream_cat: Isnt here another way? :pray:

File: packages/components/src/components/buttons/ButtonGroup/ButtonGroup.tsx [hunk 2]
```diff
@@ -39,17 +41,28 @@ const Container = styled.div<{ $variant?: Exclude<ButtonVariant, 'danger'> }>`
 `;
 
 const checkChildren = (children: Array<React.ReactNode>) =>
```
Comment (peter-sanderson) [inline packages/components/src/components/buttons/ButtonGroup/ButtonGroup.tsx]: nit: maybe rename to something like: `isValidChildrenElement` ?

File: packages/components/src/components/buttons/ButtonGroup/ButtonGroup.tsx [hunk 3]
```diff
@@ -39,17 +41,28 @@ const Container = styled.div<{ $variant?: Exclude<ButtonVariant, 'danger'> }>`
...
-    children.every(
-        child =>
-            React.isValidElement(child) && (child.type === Button || child.type === IconButton),
-    );
+    children.every(child => {
+        if (React.isValidElement(child)) {
+            if (child.type === Button || child.type === IconButton) {
```
Comment (peter-sanderson) [inline packages/components/src/components/buttons/ButtonGroup/ButtonGroup.tsx:47]: nit: does `[Button, IconButton].includes(child.type)` works. It yes, we can put them in array `const  ALLOWED_NODES = [...]`

File: packages/components/src/components/buttons/ButtonGroup/ButtonGroup.tsx [hunk 4]
```diff
@@ -63,19 +76,41 @@ export const ButtonGroup = ({
 
     if (!areChildrenValid) {
         console.error(
-            'Invalid children passed to ButtonGroup. Only Button and IconButton are allowed.',
+            'Invalid children passed to ButtonGroup. Only Button, IconButton, and Tooltip (containing Button or IconButton) are allowed.',
```
Comment (peter-sanderson) [inline packages/components/src/components/buttons/ButtonGroup/ButtonGroup.tsx:82]: nit: -> and then here do `.... ${ALLOWED_NODES.joint(",")} and Tooltip...`

File: packages/components/src/components/buttons/ButtonGroup/ButtonGroup.tsx [hunk 5]
```diff
@@ -63,19 +76,41 @@ export const ButtonGroup = ({
...
-    );
+    const childrenWithProps = React.Children.map(children, child => {
+        if (React.isValidElement(child)) {
+            if (
+                child.type === Tooltip &&
+                React.isValidElement((child.props as TooltipProps).children)
+            ) {
+                const tooltipProps = child.props as TooltipProps;
+                const tooltipChild = tooltipProps.children as React.ReactElement;
+                const tooltipChildProps: ButtonProps | IconButtonProps = tooltipChild?.props;
```
Comment (peter-sanderson) [inline packages/components/src/components/buttons/ButtonGroup/ButtonGroup.tsx]: lets extract this to ;AllowedChildrenPropsType` so it easier to add another one and it at one place

File: packages/components/src/components/buttons/ButtonGroup/ButtonGroup.tsx [hunk 6]
```diff
@@ -63,19 +76,41 @@ export const ButtonGroup = ({
...
+                const childWithProps = React.cloneElement(tooltipChild, {
+                    variant: tooltipChildProps.variant || variant,
+                    size: tooltipChildProps.size || size,
+                    isDisabled: tooltipChildProps.isDisabled || isDisabled,
+                });
+
+                return React.cloneElement(child, {}, childWithProps);
+            }
+
+            const childProps = child.props as ButtonProps | IconButtonProps;
```
Comment (peter-sanderson) [inline packages/components/src/components/buttons/ButtonGroup/ButtonGroup.tsx]: same here, with the type

File: packages/components/src/components/buttons/ButtonGroup/ButtonGroup.tsx (hunk 3 above)
Comment (jvaclavik) [reply packages/components/src/components/buttons/ButtonGroup/ButtonGroup.tsx:47]: it doesn't work here

```
Argument of type 'string | JSXElementConstructor<any>' is not assignable to parameter of type '(({ variant, size, isDisabled, isLoading, isFullWidth, icon, iconSize, iconAlignment, type, children, margin, ...rest }: ButtonProps) => Element) | (({ icon, label, bottomLabel, variant, size, iconSize, isDisabled, isLoading, ...rest }: IconButtonProps) => Element)'.
  Type 'string' is not assignable to type '(({ variant, size, isDisabled, isLoading, isFullWidth, icon, iconSize, iconAlignment, type, children, margin, ...rest }: ButtonProps) => Element) | (({ icon, label, bottomLabel, variant, size, iconSize, isDisabled, isLoading, ...rest }: IconButtonProps) => Element)'.ts(2345)
```

Comment (peter-sanderson) [reply packages/components/src/components/buttons/ButtonGroup/ButtonGroup.tsx:47]: :(

**Review by peter-sanderson — APPROVED** [review]

**Pull request merged**